front/src/common/IntervalsDataViz/Scales.tsx

```diff
@@ -0,0 +1,99 @@
+import React, { useEffect, useState } from 'react';
+import cx from 'classnames';
+
+import { roundNumber, shortNumber } from './utils';
+
+export const ResizingScale = ({
+  begin,
+  end,
+  className,
+}: {
+  begin: number;
+  end: number;
+  className?: string;
+}) => {
+  const [ticksCount, setTicksCount] = useState<number>(10);
+
+  const inf = roundNumber(begin, true);
+  const sup = roundNumber(end, false);
+  const step = roundNumber((end - begin) / ticksCount / 2, true);
+
```

Comment:
Same, you can use useDebounce or useDebouncedFunc in utils/helpers

Reply:
I think I cannot do this here, as I use window.resize event, what do you think ?

Reply:
indeed, helpers are not fitted. A general function for resize on scale should be useful